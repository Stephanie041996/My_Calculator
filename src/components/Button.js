import React from 'react';
import PropTypes from 'prop-types';

function Button({
  name, handleClick, wide, color,
}) {
  return (
    <>
      <button
        type="button"
        name={name}
        onClick={handleClick}
        style={{
          background: color || '#f5913e',
          width: wide ? '50%' : '25%',
        }}
      >
        {name}
      </button>
    </>
  );
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
  color: PropTypes.string,
  wide: PropTypes.string,

};
Button.defaultProps = {
  wide: '25%',
  color: 'color',
};

export default Button;
