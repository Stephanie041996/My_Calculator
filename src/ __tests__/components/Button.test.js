import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../components/Button';

describe('Button', () => {
  it('should renders Button', () => {
    const clickHandler = () => '1';
    const tree = renderer.create(<Button name="1" handleClick={clickHandler} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
