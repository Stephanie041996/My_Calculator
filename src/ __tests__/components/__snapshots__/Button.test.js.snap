// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should renders Button 1`] = `
<button
  name="1"
  onClick={[Function]}
  style={
    Object {
      "background": "color",
      "width": "50%",
    }
  }
  type="button"
>
  1
</button>
`;
